Development environment/Laravel Homestead setup.md

```diff
@@ -0,0 +1,153 @@
+By default IP address is defined in `Homestead.yml` but you can change it to some other address.
+
+Note for Windows users: You'll need to change permissions on `hosts` file to allow current user write access to that file.
+
+Last thing for basic configuration is database property. With this we define all databases that will be created on server startup or reprovision.
+```yml
+databases:
+  - homestead
+  - wordpress
+  ```
+
+Other thing to note is that default Homestead database name is `homestead` and password is `secret`.
+
+## Using Homestead
+
+When we are done with Homestead configuration, we can start the server with command `vagrant up`, stop the server with `vagrant halt`, restart it with `vagrant reload`. If something is changed in `Homestead.yml` configuration server will need to be reprovisioned so that all changes in configuration are reflected on our sites. To reprovision a server use `vagrant reload --provision`.
+
+Now you have everything to run your WordPress site. Download the instance of WordPress in your preferred way and set it up in `wp-config.php`. To finish installation go to your local site URL (eg. `wordpress.test`).
+
+## Extra features
+
+### Mailhog
+
+Mailhog is really useful tool for intercepting incoming and outgoing emails on local machine. To set it up first create `.env` file in `/homestead` directory and place Mailhog config inside:
+```
+MAIL_DRIVER=smtp
+MAIL_HOST=localhost
+MAIL_PORT=1025
+MAIL_USERNAME=null
+MAIL_PASSWORD=null
+MAIL_ENCRYPTION=null
+```
+After that Mailhog interface is accessible on `http://localhost:8025`
+
+### Accessing Vagrant from inside
+
+Sometimes you'll need to change some things inside the virtual machine eg. access your MySQL database, edit `php.ini` file or run some WP CLI functions. To access it type `vagrant ssh` and you are in. Since server and database are inside the virtual machine this is the only way to access them.
+
+You will be positioned in `/home` directory that will contain only `/homestead` directory (the same one that holds Homestead configuration and sites). From there you can navigate to the root folder and other parts of the Linux installation.
+
+### Sharing your sites
+
+Vagrant offers a way to temporarily make your local site publicly accessible. That can work with command `vagrant share`, but will only work if only one site is set up for that Vagrant instance.
+Homestead has its own implementation of that feature that supports sharing of multiple sites. It uses Ngrok to tunnel to your local machine.
+
+To do this first access your virtual machine with `vagrant ssh` and then run `share sitename.domain`. Ngrok will do its thing and provide you with publicly accessible URLs for you local site.
+
+For all other advanced feature please check out the [official documentation](https://laravel.com/docs/homestead)
```

Comment:
Sometimes, you'll need to change some things inside the virtual machine, e.g. access your MySQL database, edit the `php.ini` file, or run some WP CLI functions. To access it, type `vagrant ssh`, and you are in. Since the server and database are inside the virtual machine, this is the only way to access them.

You will be positioned in the `/home` directory that will contain only the `/homestead` directory (the same one that holds the Homestead configuration and sites). From there, you can navigate to the root folder and other parts of the Linux installation.

### Sharing your sites

Vagrant offers a way to temporarily make your local site publicly accessible. That can be done with the `vagrant share` command, but it will work only if only one site is set up for that Vagrant instance.
Homestead has its own implementation of that feature that supports sharing multiple sites. It uses Ngrok to tunnel to your local machine.

To do this, first access your virtual machine with `vagrant ssh`, and then run `share sitename.domain`. Ngrok will do its thing and provide you with publicly accessible URLs for you local site.

For all other advanced features, please check out the [official documentation](https://laravel.com/docs/homestead).